### DISCUSSION TITLE
Clarification: STDIN is not available for data input in a Textual application, right?

### DISCUSSION THREAD

**Discussion by nobodyinperson** [Textualize/textual#3818] [Q&A] [answered]
Hi everyone,

Just for clarification: Textual needs STDIN to get keyboard and mouse events. So one cannot use STDIN to e.g. stream data into a Textual application, right?

What would be the closest workaround? Using a different file descriptor? How wold one instruct the calling shell to shove the output of a data-producing command e.g. into file descriptor 3 and how would one read that from Python, then? 🤔

Cheers,
Yann

**Comment by willmcgugan** [comment, marked as answer]
You could do that prior to running the app, by reading STDIN in to memory or copying it to a temporary file.

Alternatively your Textual app read data from a subprocess.

**Reply by nobodyinperson** [reply]
Thanks Will, I figured as much. We're now using a subprocess and read data from there, this works fine!

We can't read the data prior to running the app, because then the app would never start. It's for streaming data from a measurement device (and subsequent plotting with textual-plotext 😉), so the data needs to be read throughout the lifetime of the app.
